### PR TITLE
Resolve #49 'Implement functionality for dashboard front end route'

### LINKED ISSUE #49: Implement functionality for `/dashboard` front-end route
Market status info can be retrieved from the API: `["response"]["TrsrBondMkt"]` yields:

1. `opngDtTm`: opening datetime
2. `clsgDtTm`: closing datetime
3. `qtnDtTm`: ? (for the love of God stop aggressively abbreviating things!!). anyways, this should be of no use to us
4. `sts`: market status ("Aberto" or "Fechado")

We can format those as:
'Mercado aberto/Mercado fechado: [current time in HHhMM]'
'Horário de funcionamento: de HHhMM às HHhMM.'

Back when I draw this view, I was thinking about having the 'Atualizado às XX/XX/XXXX, XX:XX.' with a refresh button to its left, allowing the user to refresh the data without reloading the whole page. This might be a bit unnecessary since treasury bonds rates don't fluctuate too heavily within a day.
This simplified version cleans up some information and space, concentrating information tightly (without going overboard as well).

The cards are missing all of their functionality as well. This is where the users can subscribe to/create notifications.

_Extra_: The grey circled 'i' would flip the card and exhibit information about the bond, but that would be even harder. However, it is a reasonable solution since the text is rather long. I'm still not sure what to make of it. To start off, we might leave it as a simple tooltip text.

_Extra_: it would _still_ be nice if the dashboard automatically updated the rates every 15 minutes. muh SPA

### PR DESCRIPTION
**Pull request by AtilioA**

### CODE REVIEW

File: server/src/@types/express.d.ts [hunk 1]
```diff
@@ -1,6 +1,16 @@
 // Should be as global https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express-serve-static-core/index.d.ts#L15-L23
-declare namespace Express {
-  export interface Request {
-    user: { id: string };
+declare global {
+  declare namespace Express {
+    declare interface Request {
+      user: { id: string };
+      bony: unknown;
```
Comment (AtilioA) [inline server/src/@types/express.d.ts]: :unicorn:

File: web/src/pages/Dashboard/index.tsx [hunk 2]
```diff
@@ -32,27 +112,34 @@ export default function Dashboard() {
...
+              {bonds?.PREFIXADO.map(bond => (
+                <Card key={bond.id} {...bond} />
+              ))}
             </div>
           </div>
           <div id="postfixed">
             <h1>Títulos pós-fixados</h1>
             <div id="card-list">
-              <Card />
+              {bonds?.SELIC.map(bond => (
```
Comment (AtilioA) [inline web/src/pages/Dashboard/index.tsx]: You know what, listing only SELIC bonds under "Pós-fixado" is weird. Kindly change that to "Títulos SELIC".

File: web/src/pages/Dashboard/index.tsx [hunk 3]
```diff
@@ -32,27 +112,34 @@ export default function Dashboard() {
...
+                <Card key={bond.id} {...bond} />
+              ))}
+            </div>
+          </div>
+          <div id="IGPN">
```
Comment (AtilioA) [inline web/src/pages/Dashboard/index.tsx]: "Emitidos somente até 2006, esses títulos têm vencimento marcado para o primeiro dia do ano de 2031."
I'd say we can safely ignore IGP-M bonds as they aren't circulating in the market anymore. They can only be "leveraged" through investment funds that have purchased them long ago. I don't think they'll come back as IPCA is a much more common inflation index nowadays.
In short, don't list or create a section for IGP-M bonds

File: web/src/pages/Dashboard/index.tsx [hunk 4]
```diff
@@ -32,27 +112,34 @@ export default function Dashboard() {
...
-              <Card />
-              <Card />
-              <Card />
-              <Card />
-              <Card />
-              <Card />
+              {bonds &&
+                [...bonds.IPCA, ...bonds.IGPN].map(bond => (
+                  <Card key={bond.id} {...bond} />
+                ))}
```
Comment (AtilioA) [inline web/src/pages/Dashboard/index.tsx]: (See comments for lines below/in chronological order before this one)
You can actually leave this IGP-M map here if you think this won't change the behavior of the application if there are no IGP-M bonds to be listed (in the market, today, there aren't any), and that it won't break if there are any.

File: web/src/pages/Dashboard/index.tsx [hunk 5]
```diff
@@ -1,9 +1,88 @@
...
+  function filterBonds(t: TreasuryBond[], code: number): TreasuryBond[] {
+    return t.filter(
+      b =>
+        (b.lastDateOfNegotiation === null ||
+          b.lastDateOfNegotiation === undefined) &&
+        b.indexedTo?.code === code,
+    );
+  }
+
+  async function fetchlyTreasury() {
```
Comment (AtilioA) [inline web/src/pages/Dashboard/index.tsx:48]: :man_cartwheeling:

File: web/src/pages/Dashboard/index.tsx [hunk 6]
```diff
@@ -1,9 +1,88 @@
...
 import { FiCheckCircle, FiClock, FiRefreshCw } from 'react-icons/fi';
+import axios from 'axios';
+import type { AxiosError } from 'axios';
 import { Container, BondList } from './styles';
 import Card from '../../components/Card';
+import axiosInstance from '../../config/axios';
+import { TreasuryBond } from '../../@types/global';
+
+/// Codes for each known index type
+const KnownIndexes = {
```
Comment (AtilioA) [inline web/src/pages/Dashboard/index.tsx:11]: :fire: faia `enum`

File: web/src/config/axios.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+import axios from 'axios';
+import loadRequiredEnv from '../utils/loadRequiredEnv';
+
+const axiosInstance = axios.create({
+  baseURL: loadRequiredEnv('REACT_APP_BACKEND_BASE_URL'),
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+const token = localStorage.getItem('@AlertaDoTesouro:token');
```
Comment (AtilioA) [inline web/src/config/axios.ts:11]: :100:

File: web/src/components/Card/index.tsx [hunk 8]
```diff
@@ -30,9 +83,44 @@ export default function Card() {
         <div id="card-notify">
           <p>Notifique-me quando o rendimento for</p>
           <label>
-            [MAIOR/MENOR] que <input type="number" placeholder="XX.XX" />%
+            {/* a radio selectio with option 'MAIOR' and 'MENOR' */}
```
Comment (AtilioA) [inline web/src/components/Card/index.tsx]: :goggles:

File: web/src/components/Card/index.tsx [hunk 9]
```diff
@@ -1,27 +1,80 @@
...
+  const [trigger, setTrigger] = useState<number>(0);
+  const [notificationType, setNotificationType] =
+    useState<NotificationType>('maior');
+
+  async function handleAddNotification() {
+    setCardControls({ loading: true, error: undefined });
+    try {
+      const res = await axiosInstance.post('/notifications', {
+        bond: { id },
+        value: 1.2,
```
Comment (AtilioA) [inline web/src/components/Card/index.tsx]: :eye:

File: web/src/pages/Dashboard/index.tsx (hunk 6 above)
Comment (Henriquelay) [reply web/src/pages/Dashboard/index.tsx:11]: Boa sorte usando enum fiquei 4 horas nessa porra ai cusão é isso que tava dando aquela zuada falei q tive q fazer atrocidades e vc tá aí lendo código vsf e nem terminei o bgl

Comment (AtilioA) [reply web/src/pages/Dashboard/index.tsx:11]: kkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkk falei nada ..

File: web/src/pages/Dashboard/index.tsx (hunk 4 above)
Comment (Henriquelay) [reply web/src/pages/Dashboard/index.tsx]: Well there was one yesterday you twat

File: web/src/components/Card/index.tsx (hunk 9 above)
Comment (Henriquelay) [reply web/src/components/Card/index.tsx]: LITERALMENTE NÃO TÁ ACABADO CU DE FUZIL

Comment (Henriquelay) [reply web/src/components/Card/index.tsx]: E SE TESTASSE IA VER QUE NOTIFICA NEM TÁ DANDO CERTO

Comment (AtilioA) [reply web/src/components/Card/index.tsx]: kjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjjj eu sei porra para

File: web/src/pages/Dashboard/index.tsx (hunk 4 above)
Comment (AtilioA) [reply web/src/pages/Dashboard/index.tsx]: There's only one and it's:tm: not being traded anymore

Comment (Henriquelay) [reply web/src/pages/Dashboard/index.tsx]: E vc acha q eu ia sabe disso como? Eu vejo na API eu coloco no painel 🔧

Comment (AtilioA) [reply web/src/pages/Dashboard/index.tsx]: literalmente estou te falando!

File: web/src/pages/Dashboard/index.tsx (hunk 2 above)
Comment (AtilioA) [reply web/src/pages/Dashboard/index.tsx]: !

File: web/src/pages/Dashboard/index.tsx [hunk 10]
```diff
@@ -32,27 +110,25 @@ export default function Dashboard() {
...
-              <Card />
-              <Card />
-              <Card />
+              {bonds?.PREFIXADO.map(bond => (
+                <Card key={bond.id} {...bond} />
+              ))}
             </div>
           </div>
           <div id="postfixed">
             <h1>Títulos pós-fixados</h1>
```
Comment (henrique-vaultree) [inline web/src/pages/Dashboard/index.tsx:127]:
```suggestion
            <h1>Títulos SELIC</h1>
```

Comment (AtilioA) [reply web/src/pages/Dashboard/index.tsx:127]: image
vuei

Comment (AtilioA) [inline web/src/pages/Dashboard/index.tsx]: não tô conseguindo alterar pela review não, n lembro como faz

Comment (Henriquelay) [inline web/src/pages/Dashboard/index.tsx]:
```suggestion
            <h1>Títulos SELIC</h1>
```

**Comment by AtilioA** [comment]
1985

**Review by AtilioA — COMMENTED** [review]
LGTM :+1: (I can't approve since I opened the PR)

**Comment by Henriquelay** [comment]
You just shouldn't have

**Comment by AtilioA** [comment]
> You just shouldn't have

:-1:

**Pull request merged**